### PR TITLE
protodoc: Use jinja2 for rendering proto docs

### LINKED ISSUE #15351: Use jinja for protodoc
### description

the complexity of protodoc seems to have outgrown the python templating pattern used

this is resulting in a lot of `\n` and `{indent}` fu and making it increasingly difficult to work with

one possible solution to this problem is to use python only to create structured variables that a proper templating/markup lang like jinja can consume

### PR DESCRIPTION
**Pull request by phlax**
Commit Message: protodoc: Use jinja2 for rendering proto docs
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Partial fix for #15351
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
this PR only adds jinja and moves the current explicit templates

i think we could go quite a bit further, but i think this is a good start

**Comment by moderation** [comment]
Looks good and @htuch should take a quick look

File: tools/protodoc/protodoc.py [hunk 1]
```diff
@@ -222,31 +227,19 @@ def FormatExtension(extension):
...
         f"\n\nDid you forget to add '{extension}' to source/extensions/extensions_build_config.bzl?\n\n"
     )
     exit(1)  # Raising the error buries the above message in tracebacks.
 
-
-def FormatExtensionList(items, prefix="extension", indent=2):
-  indent = " " * indent
-  formatted_list = "\n".join(f"{indent}- :ref:`{ext} <{prefix}_{ext}>`" for ext in items)
-  return f"{formatted_list}\n{indent}\n"
+  return EXTENSION_TEMPLATE.render(extension=extension,
```
Comment (htuch) [inline tools/protodoc/protodoc.py:239]: There are probably a ton of other cleanup possibilities in this file. Would you prefer to leave them as TODOs?

**Review by htuch — COMMENTED** [review]
Thanks, much cleaner!

File: tools/protodoc/protodoc.py (hunk 1 above)
Comment (phlax) [reply tools/protodoc/protodoc.py:239]: yep - for now i think yes - im happy to follow up tho

Comment (phlax) [reply tools/protodoc/protodoc.py:239]: i updated the PR to "partial fix for.," so we keep the sphinx ticket open once this lands

**Review by htuch — APPROVED** [review]
/lgtm deps
LGTM, thanks!

**Pull request merged**